modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/CheckboxTag.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class CheckboxTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		return super.doStartTag();
+	}
+
+	public boolean getChecked() {
+		return _checked;
+	}
+
+	public boolean getDisabled() {
+		return _disabled;
+	}
+
+	public boolean getIndeterminate() {
+		return _indeterminate;
+	}
+
+	public String getLabel() {
+		return _label;
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public boolean getShowLabel() {
+		return _showLabel;
+	}
+
+	public String getValue() {
+		return _value;
+	}
+
+	public void setChecked(boolean checked) {
+		_checked = checked;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setIndeterminate(boolean indeterminate) {
+		_indeterminate = indeterminate;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setShowLabel(boolean showLabel) {
```

Comment:
Do we really need `showLabel`? Can you check if we migrated other similar components and kept this pattern?

I think having `label` should be enough and we can check for `Validator.isNotNull(label)` so you can simply not have a label.

The usual case here is:

```
<clay:checkbox
    label="The Label"
    showLabel="<%= someCondition %>"
/>

vs

<clay:checkbox 
    label='<%= someCondition ? "The Label" : StringPool.BLANK %>'
/>
```

Reply:
I agree it should be enough, I kept it in because I wasn't sure if I can get rid of it, now I know! 👍

Reply:
@jbalsas @carloslancha I checked similar components and this pattern, and saw that [RadioTag.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/RadioTag.java#L153-L155) uses it to set `sr-only` class to the element when it's set to `false`.

Reply:
Not sure what to do then with this, should i make it so that it still has the `showLabel` attribute, or not 🤷

Reply:
We have to keep the same API and I think this one is important so I'd say that:

- You can add that `sr-only` class to the label in you server side markup
- We need to add that option in ClayCheckbox so you can pass it to the react component.